src/snowflake/connector/pandas_tools.py

```diff
@@ -152,37 +152,27 @@ def write_pandas(
         )
 
     if quote_identifiers:
-        location = (
-            (('"' + database + '".') if database else "")
-            + (('"' + schema + '".') if schema else "")
-            + ('"' + table_name + '"')
+        location = (('"' + database + '".') if database else "") + (
+            ('"' + schema + '".') if schema else ""
         )
     else:
-        location = (
-            (database + "." if database else "")
-            + (schema + "." if schema else "")
-            + (table_name)
+        location = (database + "." if database else "") + (
+            schema + "." if schema else ""
         )
     if chunk_size is None:
         chunk_size = len(df)
+
+    def random_name(length: int = 10) -> str:
```

Comment:
It will be hard with my change. Because I need to create a table with random name, and when I use previous, if the generated random name is the same as the target table name, because our SQL to create such a table is `CREATE TABLE IF EXISTS`, the intermediate table with random name will not be created if the table already exists. So I feel this change is needed and it's actually safe since I just remove the while loop and make the name more random.

Reply:
I see. It's probably an overkill but we could remove `IF EXISTS` and use a while-loop as well, but I don't think it is strictly necessary if we are confident that there won't be conflict.

Reply:
done, I moved this method to util_text.py